src/planner/binder/query_node/bind_select_node.cpp

```diff
@@ -48,6 +48,25 @@ unique_ptr<Expression> Binder::BindDelimiter(ClientContext &context, unique_ptr<
 	return expr;
 }
 
+unique_ptr<Expression> Binder::BindDelimiter(ClientContext &context, unique_ptr<ParsedExpression> delimiter,
+                                             double &delimiter_value) {
+	auto new_binder = Binder::CreateBinder(context, this, true);
+	ExpressionBinder expr_binder(*new_binder, context);
+	expr_binder.target_type = LogicalType::DOUBLE;
+	auto expr = expr_binder.Bind(delimiter);
+	if (expr->IsFoldable()) {
```

Comment:
This is exactly the same code as `Binder::BindDelimiter<int64_t>` but with a minor change that the LogicalType is different, could you turn this into a templated function instead to avoid unnecessary code duplication?